pkg/util/serialization/deserialization_util.go

```diff
@@ -36,8 +36,18 @@ func NewPosAndBuf(col *chunk.Column, idx int) PosAndBuf {
 		Pos: &pos}
 }
 
-func deserializeLength(posAndBuf PosAndBuf) int {
-	return DeserializeInt(posAndBuf)
+func deserializeBuffer(posAndBuf PosAndBuf) []byte {
+	bufLen := DeserializeInt(posAndBuf) // Get buffer length
+	retVal := posAndBuf.Buf[*posAndBuf.Pos : *posAndBuf.Pos+int64(bufLen)]
+	*posAndBuf.Pos += int64(bufLen)
+	return retVal
+}
+
+// DeserializeByte deserializes byte type
+func DeserializeByte(posAndBuf PosAndBuf) byte {
```

Comment:
deserializeByte

Reply:
> deserializeByte

I think this should be a public though it has not been used yet.